### ISSUE TITLE
Bring back reading settings from data attributes

### ISSUE THREAD

**Issue by benpickles** [benpickles/peity#49]
https://github.com/benpickles/peity/commit/37849cd320f374aaa7b6d61fe90b1c47986c44dd#commitcomment-8940091

**Comment by benpickles** [comment]
I've discovered a problem with this feature. Unfortunately, and for seemingly no good reason, jQuery modifies the keys of data attributes that it reads - for instance `stroke-width` gets transformed to `strokeWidth`. This means that by just passing the data object to Peity some intended settings won't be applied.

http://jsfiddle.net/benpickles/nej143wa/

**Comment by asafbrukarz** [comment]
Yes, as per jQuery's [documentation](http://api.jquery.com/data/#data-html5) this was changed in v1.6 to conform to W3C specification. 

Having a look at your code, however, it seems that most variables are camel cased, so your variable is actually named strokeWidth. Where do you expect to see a problem in merging camel cased names?

**Comment by benpickles** [comment]
Thanks, I didn't know that.

Yes, you're right. It's not a big issue, I guess it just surprised me.

**Comment by benpickles** [comment]
Take a look at this: https://github.com/benpickles/peity/compare/read-data-attributes

What do you think?

**Comment by asafbrukarz** [comment]
I think that's a sensible approach.

As all of the options for a chart sit within one element, there's also the option of configuring the defaults using something like: 

```
<span class="pie" data-peity-options='{"fill": ["red", "#eeeeee"], "innerRadius": 4 }'></span>
```

This would save you from having to normalise any data coming from the prefix-ed attributes.

I don't have a preference either way though.

[Source](http://markdalgleish.com/2011/05/creating-highly-configurable-jquery-plugins/)

**Comment by benpickles** [comment]
That is a much better idea :smile:

**Comment by benpickles** [comment]
Thanks for that, it works much better.

https://github.com/benpickles/peity/compare/read-data-attributes

**Comment by asafbrukarz** [comment]
Nice. And with only a few hundred bytes growth.

**Comment by benpickles** [comment]
I've released version 3.1 with this feature. Thanks for your time and effort on this, it's interaction like this that makes open source worth doing.
